lib/ansible/plugins/inventory/foreman.py

```diff
@@ -175,7 +175,7 @@ def _get_all_params_by_id(self, hid):
         url = "%s/api/v2/hosts/%s" % (self.foreman_url, hid)
         ret = self._get_json(url, [404])
         if not ret or not isinstance(ret, MutableMapping) or not ret.get('all_parameters', False):
```

Comment:
Just noticed. Not a blocker, but I wonder whether we can avoid accessing the `ret` dict twice and do:
```suggestion
         if not ret or not isinstance(ret, MutableMapping):
```
and then on line 179: `return ret.get('all_parameters', {})`?

Reply:
But that can be addressed in a separate PR :)

Reply:
I'll put that in another PR so that we can get this bugfix in and backported.

Reply:
:+1: